crates/ruff_linter/src/rules/flake8_pyi/rules/redundant_none_literal.rs

```diff
@@ -0,0 +1,101 @@
+use ruff_diagnostics::{Applicability, Diagnostic, Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::Expr;
+use ruff_python_semantic::analyze::typing::traverse_literal;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for redundant `Literal[None]` annotations.
+///
+/// ## Why is this bad?
+/// The `Literal[None]` annotation is equivalent to `None`, as the type `None` consists
+/// of just a single value.
```

Comment:
The motivation of this rule is purely for stylistic consistency: it doesn't detect a bug or anything that's incorrect, it just makes your annotations more concise and enforces a consistent style. I'd prefer to say that more explicitly here, like we do in [the flake8-pyi docs](https://github.com/PyCQA/flake8-pyi/blob/main/ERRORCODES.md#Y061):

```suggestion
/// `Literal[None]` is legal as a type annotation, but means the same thing as `None`.
/// For stylistic consistency, prefer using `None`, which is more concise. 
```